cpp/include/ucxx/worker_progress_thread.h

```diff
@@ -47,8 +47,8 @@ typedef void* ProgressThreadStartCallbackArg;
  */
 class WorkerProgressThread {
  private:
-  std::thread _thread{};     ///< Thread object
-  bool _stop{false};         ///< Signal to stop on next iteration
+  std::thread _thread{};                                       ///< Thread object
+  std::shared_ptr<bool> _stop{std::make_shared<bool>(false)};  ///< Signal to stop on next iteration
```

Comment:
Why does this now need to be a shared ptr? Is it that because it was passed around by reference between threads previously, it might have gone out of scope on thread A when read on thread B?

Reply:
Initially I thought so, but while debugging I observed `std::ref` containing garbage briefly after starting the thread. I couldn't figure what was happening, the parent was still alive and `std::shared_ptr` resolved the issue so I just decided not to spend much more time on it.

Reply:
Fair.